omnigibson/objects/controllable_object.py

```diff
@@ -92,9 +93,19 @@ def __init__(
         self.dof_names_ordered = None
         self._control_enabled = True
 
+        if relative_prim_path:
+            # If prim path is specified, assert that the last element starts with controllable_ to ensure that
+            # the object will be included in the ControllableObjectViewAPI.
+            assert relative_prim_path.split("/")[-1].startswith(
+                "controllable_"
+            ), "If relative_prim_path is specified, the last element of the path must start with 'controllable_'."
```

Comment:
hmm seems a bit hacky, do we have to do raw string matching for `ControllableObjectViewAPI`?

Reply:
@cgokmen

Reply:
+1 to what eric said. But I think in order to do basic glob regex matching, we have to do string matching...not entirely sure tho. Would be ideal if we could think of a slightly better way (Alternatively, we could have some internal convention to denote that this is a "key" string -- e.g.: `controllable:<NAME>` so differentiate between the desired name and the ultimate name passed into IS

Reply:
and we should also have some non alphanumeric characters too included anyways to avoid accidental naming collisions from users naively setting "controllable" as part of an obj name to an obj that is _not_ controllable

Reply:
This has to be a string match correct, it's the only way. Re: the special string, I think USD is pretty picky in terms of what you can pick in a prim path. I don't know what's the appropriate place where we should put an assertion in the prim base somewhere to check that an object's prim path doesn't contain "controllable_" or the dummy's special string. I think the best approach is making sure that in the actual view class after globbing the number of objects that match the string are equal to the number of controllableobject instances or dummies in the scene. If that assertion fails we can output a message that tells the user to watch out for name collisions with the special strings.